js/src/Map.js

```diff
@@ -2179,6 +2182,51 @@ function save_svg() {
     }.bind(this));
 }
 
+function save_png() {
+    /** Rescale the canvas and save png image.
+
+     */
+    // run the before callback
+    this.callback_manager.run('before_png_export');
+
+    // turn of zoom and translate so that illustrator likes the map
+    var window_scale = this.zoom_container.window_scale,
```

Comment:
there is quite a bit of repeated code here. can you think about a nice way to wrap `save_png` and `save_svg` in a parent function?